tests/test_app.py

```diff
@@ -1,45 +1,111 @@
 import os
+from pprint import pprint
+
+from ncpi_fhir_utility import cli
+from ncpi_fhir_utility.loader import load_resources
 
 import pytest
 from click.testing import CliRunner
-
-from ncpi_fhir_utility import cli
 from conftest import (
-    PROFILE_DIR,
-    EXAMPLE_DIR,
-    EXTENSION_DIR,
-    copy_resources_into_ig
+    RESOURCE_DIR,
+    FHIR_API,
+    FHIR_USER,
+    FHIR_PW
 )
+SEARCH_PARAM_DIR = os.path.join(RESOURCE_DIR, 'search')
+ASIAN = '2028-9'
+NON_HISPANIC = '2186-5'
 
 
-VALID_PROFILE_DIR = os.path.join(PROFILE_DIR, 'valid')
-VALID_PROFILES = [os.path.join(VALID_PROFILE_DIR, f)
-                  for f in os.listdir(VALID_PROFILE_DIR)]
+@pytest.mark.parametrize(
+    "resource_dir",
+    [
+        (os.path.join(RESOURCE_DIR, 'terminology')),
+        (os.path.join(RESOURCE_DIR, 'extensions')),
+        (os.path.join(RESOURCE_DIR, 'profiles')),
+        (os.path.join(RESOURCE_DIR, 'search')),
+        # TODO - This fails because we don't have the external terminology
+        # server setup with our integration test server.
+        # Once that is complete, then uncomment the line below
+        # (os.path.join(RESOURCE_DIR, 'examples'),
+    ]
+)
+def test_deploy_model(resource_dir):
+    """
+    Test loading the FHIR model into the integration test server
+
+    Order of load: terminology, extensions, profiles, search parameters,
+    example resources.
+    """
+    runner = CliRunner()
+    result = runner.invoke(
+        cli.publish,
+        [resource_dir, '--base_url', FHIR_API,
+         '--username', FHIR_USER, '--password', FHIR_PW]
+    )
+    assert result.exit_code == 0
 
 
 @pytest.mark.parametrize(
-    "dir_list, expected_code",
+    "resource_type,expected_count",
     [
-        ([os.path.join(PROFILE_DIR, 'valid'),
-          os.path.join(EXAMPLE_DIR, 'valid'),
-          os.path.join(EXTENSION_DIR, 'valid')], 0),
-        ([os.path.join(PROFILE_DIR, 'invalid'),
-          os.path.join(EXAMPLE_DIR, 'invalid')], 1),
-    ],
+        ('Patient', 10),
+        # Add additional tuples of the form: (resource_type, expected_count)
+    ]
 )
-def test_ig_validation(temp_site_root, dir_list, expected_code):
+def test_ingest_plugin(
```

Comment:
It fails with 401 since when `kids_first_fhir.py` called, the following envs are all reset to `None`:

```python
FHIR_USER = os.getenv("FHIR_USER")
FHIR_PW = os.getenv("FHIR_PW")
```

Reply:
Yea that's expected. The developer should set those environment variables before running the tests. But I like the idea of falling back to the defaults. I can add that